batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,175 @@
+//            String query = "select * from student limit ? offset ?";
+//            PreparedStatement ps = conn.prepareStatement(query);
+//            ps.setInt(1,totalDataToFetch);
+//            ps.setInt(2,offset);
+            Statement stmt = conn.createStatement();
+            ResultSet studentResult = stmt.executeQuery(query);
+
+            while(studentResult.next()){
+                Integer id = studentResult.getInt("id");
+                Integer rollNo = studentResult.getInt("roll");
+                String department = studentResult.getString("department");
+                String batch = studentResult.getString("batch");
+                String name = studentResult.getString("name");
+                String address = studentResult.getString("address");
+                Date dob = studentResult.getDate("dob");
+
+                Student student = new Student();
+                student.setId(id);
+                student.setRoll(rollNo);
+                student.setDepartment(department);
+                student.setBatch(batch);
+                student.setName(name);
+                student.setAddress(address);
+                student.setDob(dob);
+
+                studentList.add(student);
+            }
+
+            conn.close();
+            return studentList;
+
+        } catch (NamingException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        }
+    }
+
+
+    public void insert(Student stud) throws DataException {
+        try {
+            Connection conn = DbConnection.getConnection();
+
+            String studentQuery = "insert into student (name,address,dob,department,batch,roll) values(?,?,?,?,?,?)";
+            PreparedStatement ps1 = conn.prepareStatement(studentQuery);
+            ps1.setString(1,stud.getName());
+            ps1.setString(2,stud.getAddress());
+            ps1.setDate(3, new java.sql.Date(stud.getDob().getTime()));
+            ps1.setString(4,stud.getDepartment());
+            ps1.setString(5,stud.getBatch());
+            ps1.setInt(6, stud.getRoll());
+            ps1.executeUpdate();
+
+            conn.close();
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        } catch (NamingException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        }
+
+    }
+
+
+    public void delete(Integer studentId) throws DataException {
+        try {
+            Connection conn = DbConnection.getConnection();
+
+            String query = "delete from student where id=?";
+            PreparedStatement ps = conn.prepareStatement(query);
+            ps.setInt(1,studentId);
+            ps.executeUpdate();
+
+        } catch (NamingException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)